fsharp-backend/src/LibBackend/StdLib/LibJwt.fs

```diff
@@ -44,285 +48,232 @@ let varB = TVariable "b"
-//                   ~key
-//                   (Unicode_string.to_string token)
-//               with
-//             | Some (headers, payload) ->
-//                 [ ("header", Dval.of_unknown_json_v1 headers)
-//                 ; ("payload", Dval.of_unknown_json_v1 payload) ]
-//                 |> Prelude.StrDict.from_list_exn
-//                 |> DObj
-//                 |> OptJust
-//                 |> DOption
-//             | None ->
-//                 DOption OptNothing ) )
-//         | _ ->
-//             incorrectArgs ())
-//   ; sqlSpec = NotYetImplementedTODO
-//   ; previewable = Impure
-//   ; deprecated = ReplacedBy(fn "" "" 0) }
-// ; { name = fn "JWT" "verifyAndExtract" 1
-//   ; parameters = [Param.make "pemPubKey" TStr ""; Param.make "token" TStr ""]
-//   ; returnType = TResult
-//   ; description =
-//       "Verify and extra the payload and headers from an rfc751J9 JSON Web Token that uses the RS256 algorithm. Takes an unencrypted RSA public key in PEM format."
-//   ; fn =
-//         (function
-//         | _, [DStr key; DStr token] ->
-//           ( try
-//               match
-//                 key
-//                 |> Unicode_string.to_string
-//                 |> Cstruct.of_string
-//                 |> X509.Encoding.Pem.Public_key.of_pem_cstruct1
-//               with
-//               | `EC_pub _ ->
-//                   DOption OptNothing
-//               | `RSA key ->
-//                 ( match
-//                     verify_and_extract_v1
-//                       ~key
-//                       (Unicode_string.to_string token)
-//                   with
-//                 | Ok (headers, payload) ->
-//                     [ ("header", Dval.of_unknown_json_v1 headers)
-//                     ; ("payload", Dval.of_unknown_json_v1 payload) ]
-//                     |> Prelude.StrDict.from_list_exn
-//                     |> DObj
-//                     |> ResOk
-//                     |> DResult
-//                 | Error msg ->
-//                     DResult (ResError (DStr msg)) )
-//             with Invalid_argument msg ->
-//               let msg =
-//                 if msg = "No public keys" then "Invalid public key" else msg
-//               in
-//               DResult (ResError (DStr msg)) )
-//         | _ ->
-//             incorrectArgs ())
-//   ; sqlSpec = NotYetImplementedTODO
-//   ; previewable = Impure
-//   ; deprecated = NotDeprecated } ]
-//
+let signAndEncode (key : RSA) (extraHeaders : DvalMap) (payload : string) : string =
+  let header =
+    extraHeaders
+    |> Map.add "alg" (DStr "RS256")
+    |> Map.add "typ" (DStr "JWT")
+    |> DObj
+    |> DvalRepr.toPrettyMachineJsonStringV1
+    |> toBytes
+    |> base64Encode
+    |> base64ToUrlEncoded
+
+  let payload = payload |> toBytes |> base64Encode |> base64ToUrlEncoded
+
+  let body = header + "." + payload
+
+  let sha256 = SHA256.Create()
+  let hash = body |> toBytes |> sha256.ComputeHash
```

Comment:
I think this would be clearer if `hash` was inlined in the `let signature` call later